### ISSUE TITLE
0.76.0 breaks when unit tests are disabled for a variant

### ISSUE THREAD

**Issue by gabrielittner** [autonomousapps/dependency-analysis-android-gradle-plugin#452]
**Build scan link**

The reproducer of https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/issues/423 can be used after bumping the plugin version there

**Plugin version**

0.76.0

**Gradle version**

7.0 and 7.2

**(Optional) Android Gradle Plugin (AGP) version**

4.2.0 and 7.0.1

**Describe the bug**

I've disabled the release variant and get:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':settings:adviceRelease' (type 'AdvicePerVariantTask').
  - Type 'AdvicePerVariantTask' property 'testCompileGraph' specifies file '/Users/gabriel/Downloads/android-test/settings/build/reports/dependency-analysis/release/graph/graph-test-compile.json' which doesn't exist.

    Reason: An input file was expected to be present but it doesn't exist.

    Possible solutions:
      1. Make sure the file exists before the task is called.
      2. Make sure that the task which produces the file is declared as an input.
```

**To Reproduce**
1. Disable unit tests for a variant
2. Run buildHealth

**Expected behavior**

Succeeds without error.

**Comment by autonomousapps** [comment]
Thanks. I really thought I had fixed that!

**Comment by gabrielittner** [comment]
The original issue was fixed in 0.75, this one then started happening in 0.76

**Comment by autonomousapps** [comment]
The issue in 75 was actually hiding the one in 76. My regression test
just wasn't robust enough (I was being too clever).

On August 28, 2021, GitHub ***@***.***> wrote:
> The original issue was fixed in 0.75, this one then started happening
> in 0.76
>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/autonomousapps/dependency-analysis-android-gradle-
plugin/issues/452#issuecomment-907670007>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABJG5PIJM4AMAHZNVDCHYFDT7ETVVANCNFSM5C3JO2SQ>.
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-
email&mt=8&pt=524675> or Android
<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-
email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by autonomousapps** [comment]
Resolved via related PR.
